Rework Abins docs: input data formats info split out to another page

ajjackson opened the pull request:
*Migrated #38518 from external fork to a branch on mantidproject; see initial review there. Hopefully this will make it easier to get useful package artifacts from CI and verify that the docs are ok.*

### Description of work

Add a page to Concepts docs

#### Summary of work

Break out some information from Abins / Abins2D Algorithm docs and expand with a bit more information in a new Concepts doc, linked from the Algorithm docs.

#### Purpose of work

resolves #38352

A bit more documentation is needed to clarify how isotope data should be used with CASTEP calculations fed to Abins. If we put all the code-specific "gotchas" into the Algorithm docs it can

a) get a bit unwieldy;
b) lead to a lot of duplication between Abins and Abins2D docs.

The new page also provides a natural home for a quick-reference table of supported data formats.

**Report to:** Pablo Gila-Herranz


### To test:

Docs should build. Probably a good idea to look at them too: make sure the new page appears in the Concepts index as well as the Abins Algorithm pages.

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

ajjackson committed:
Rework Abins docs: input data formats info split out to another page

ajjackson committed:
Update changelog

ajjackson committed:
Remove spurious "Title" from table RST

SilkeSchomann commented:
I have tried to build a package from this branch, but it fails an Abins test: [python.scripts.AbinsPowderCalculatorTest.AbinsPowderCalculatorTest](https://builds.mantidproject.org/job/build_packages_from_branch/971/testReport/projectroot.scripts.test/Abins/python_scripts_AbinsPowderCalculatorTest_AbinsPowderCalculatorTest/)

ajjackson commented:
That's the scipy-related one from last week. I'll try rebasing.

ajjackson committed:
Fix AbinsDataFormats table header; drop spaces from label

Even without spaces in the label, I am having trouble getting the new
file to appear in the concepts index.

ajjackson commented:
I can get the docs to build on Linux, but the new entry does not appear on the Concepts index. I tried changing to a section label that doesn't contain spaces, but it doesn't seem to have helped.

I noticed that the table header still looked a bit off, that should be fixed now.

ajjackson committed:
Add concepts category to new docs file; add local contents

This category seems to be required for the new page to show up in
concepts index.

Also add a little contents section to the top of the file.

SilkeSchomann reviewed (approved):
I can confirm that the table header on the AbinsDataFormats page is now as expected and there is an entry for AbinsDataFormats on the concepts page. The links on the Abins-v1 and Abins2D-v1 pages still work 👍🏻
